### ISSUE TITLE
Silhouette Dependencies

### ISSUE THREAD

**Issue by rfranco** [mohiva/play-silhouette#255]
When added silhouette as dependency that include
"com.typesafe.play" %% "play-test" % "2.3.7" in runtime dependency.

For that reason that add alot of test jar in the dist.
Basic Play project has 28MB of jar dependencies but with silhouette that goes to 83MB.

My suggestion is split in two project, silhouette and testkit.

Basic Play Project Dependency List
```bash
265K ch.qos.logback.logback-classic-1.1.1.jar
418K ch.qos.logback.logback-core-1.1.1.jar
 36K com.fasterxml.jackson.core.jackson-annotations-2.3.2.jar
195K com.fasterxml.jackson.core.jackson-core-2.3.2.jar
894K com.fasterxml.jackson.core.jackson-databind-2.3.2.jar
2.4M com.typesafe.akka.akka-actor_2.11-2.3.4.jar
 15K com.typesafe.akka.akka-slf4j_2.11-2.3.4.jar
214K com.typesafe.config-1.2.1.jar
6.8K com.typesafe.netty.netty-http-pipelining-1.1.2.jar
 11K com.typesafe.play.build-link-2.3.7.jar
5.2K com.typesafe.play.play-datacommons_2.11-2.3.7.jar
4.5K com.typesafe.play.play-exceptions-2.3.7.jar
364K com.typesafe.play.play-functional_2.11-2.3.7.jar
714K com.typesafe.play.play-iteratees_2.11-2.3.7.jar
616K com.typesafe.play.play-json_2.11-2.3.7.jar
3.6M com.typesafe.play.play_2.11-2.3.7.jar
 59K com.typesafe.play.twirl-api_2.11-1.0.2.jar
258K commons-codec.commons-codec-1.9.jar
1.8K default.play-scala-1.0-SNAPSHOT-assets.jar
 57K default.play-scala-1.0-SNAPSHOT.jar
1.2M io.netty.netty-3.9.3.Final.jar
 15K javax.transaction.jta-1.1.jar
568K joda-time.joda-time-2.3.jar
308K org.apache.commons.commons-lang3-3.1.jar
698K org.javassist.javassist-3.18.2-GA.jar
 97K org.joda.joda-convert-1.6.jar
410K org.scala-lang.modules.scala-parser-combinators_2.11-1.0.1.jar
640K org.scala-lang.modules.scala-xml_2.11-1.0.1.jar
5.3M org.scala-lang.scala-library-2.11.4.jar
4.2M org.scala-lang.scala-reflect-2.11.1.jar
616K org.scala-stm.scala-stm_2.11-0.7.jar
 16K org.slf4j.jcl-over-slf4j-1.7.6.jar
4.8K org.slf4j.jul-to-slf4j-1.7.6.jar
 28K org.slf4j.slf4j-api-1.7.6.jar
1.3M xerces.xercesImpl-2.11.0.jar
215K xml-apis.xml-apis-1.4.01.jar
total 26.8MB
```

Basic Play Project + Silhouette Dependencies
```bash
317K cglib.cglib-nodep-2.1_3.jar
265K ch.qos.logback.logback-classic-1.1.1.jar
418K ch.qos.logback.logback-core-1.1.1.jar
 16K com.atlassian.jwt.jwt-api-1.2.1.jar
 40K com.atlassian.jwt.jwt-core-1.2.1.jar
 36K com.fasterxml.jackson.core.jackson-annotations-2.3.2.jar
195K com.fasterxml.jackson.core.jackson-core-2.3.2.jar
894K com.fasterxml.jackson.core.jackson-databind-2.3.2.jar
 32K com.google.code.findbugs.jsr305-2.0.3.jar
2.1M com.google.guava.guava-16.0.1.jar
699K com.mohiva.play-silhouette_2.11-2.0-SNAPSHOT.jar
126K com.nimbusds.nimbus-jose-jwt-2.16.jar
518K com.ning.async-http-client-1.8.14.jar
 27K com.novocode.junit-interface-0.11-RC1.jar
2.4M com.typesafe.akka.akka-actor_2.11-2.3.4.jar
 15K com.typesafe.akka.akka-slf4j_2.11-2.3.4.jar
214K com.typesafe.config-1.2.1.jar
6.8K com.typesafe.netty.netty-http-pipelining-1.1.2.jar
 11K com.typesafe.play.build-link-2.3.7.jar
 42K com.typesafe.play.play-cache_2.11-2.3.6.jar
5.2K com.typesafe.play.play-datacommons_2.11-2.3.7.jar
4.5K com.typesafe.play.play-exceptions-2.3.7.jar
364K com.typesafe.play.play-functional_2.11-2.3.7.jar
714K com.typesafe.play.play-iteratees_2.11-2.3.7.jar
616K com.typesafe.play.play-json_2.11-2.3.7.jar
156K com.typesafe.play.play-test_2.11-2.3.6.jar
586K com.typesafe.play.play-ws_2.11-2.3.6.jar
3.6M com.typesafe.play.play_2.11-2.3.7.jar
 59K com.typesafe.play.twirl-api_2.11-1.0.2.jar
258K commons-codec.commons-codec-1.9.jar
562K commons-collections.commons-collections-3.2.1.jar
170K commons-io.commons-io-2.2.jar
 61K commons-logging.commons-logging-1.1.3.jar
1.8K default.play-scala-1.0-SNAPSHOT-assets.jar
 57K default.play-scala-1.0-SNAPSHOT.jar
1.2M io.netty.netty-3.9.3.Final.jar
 62K javax.activation.activation-1.1.jar
509K javax.mail.mail-1.4.7.jar
 15K javax.transaction.jta-1.1.jar
568K joda-time.joda-time-2.3.jar
239K junit.junit-4.11.jar
985K net.java.dev.jna.jna-3.4.0.jar
892K net.java.dev.jna.platform-3.4.0.jar
2.2K net.jcip.jcip-annotations-1.0.jar
 50K net.minidev.json-smart-1.1.1.jar
1.3M net.sf.ehcache.ehcache-core-2.6.8.jar
346K net.sourceforge.cssparser.cssparser-0.9.11.jar
1.2M net.sourceforge.htmlunit.htmlunit-2.13.jar
1.0M net.sourceforge.htmlunit.htmlunit-core-js-2.13.jar
122K net.sourceforge.nekohtml.nekohtml-1.9.19.jar
6.4K oauth.signpost.signpost-commonshttp4-1.2.1.2.jar
 44K oauth.signpost.signpost-core-1.2.1.2.jar
 51K org.apache.commons.commons-exec-1.1.jar
308K org.apache.commons.commons-lang3-3.1.jar
572K org.apache.httpcomponents.httpclient-4.3.1.jar
276K org.apache.httpcomponents.httpcore-4.3.jar
 36K org.apache.httpcomponents.httpmime-4.3.1.jar
1.9M org.bouncycastle.bcprov-jdk15on-1.47.jar
 77K org.easytesting.fest-assert-1.4.jar
 30K org.easytesting.fest-util-1.1.6.jar
 94K org.eclipse.jetty.jetty-http-8.1.12.v20130726.jar
102K org.eclipse.jetty.jetty-io-8.1.12.v20130726.jar
281K org.eclipse.jetty.jetty-util-8.1.12.v20130726.jar
123K org.eclipse.jetty.jetty-websocket-8.1.12.v20130726.jar
 83K org.fluentlenium.fluentlenium-core-0.9.2.jar
 13K org.fluentlenium.fluentlenium-festassert-0.9.2.jar
 44K org.hamcrest.hamcrest-core-1.3.jar
698K org.javassist.javassist-3.18.2-GA.jar
 97K org.joda.joda-convert-1.6.jar
 37K org.json.json-20080701.jar
 17K org.mindrot.jbcrypt-0.3m.jar
1.4M org.mockito.mockito-core-1.9.5.jar
 28K org.objenesis.objenesis-1.0.jar
410K org.scala-lang.modules.scala-parser-combinators_2.11-1.0.1.jar
640K org.scala-lang.modules.scala-xml_2.11-1.0.1.jar
 13M org.scala-lang.scala-compiler-2.11.0.jar
5.3M org.scala-lang.scala-library-2.11.4.jar
4.2M org.scala-lang.scala-reflect-2.11.1.jar
 14K org.scala-sbt.test-interface-1.0.jar
616K org.scala-stm.scala-stm_2.11-0.7.jar
301K org.scalaz.scalaz-concurrent_2.11-7.0.6.jar
8.5M org.scalaz.scalaz-core_2.11-7.0.6.jar
409K org.scalaz.scalaz-effect_2.11-7.0.6.jar
304K org.seleniumhq.selenium.selenium-android-driver-2.39.0.jar
141K org.seleniumhq.selenium.selenium-api-2.39.0.jar
 11K org.seleniumhq.selenium.selenium-chrome-driver-2.39.0.jar
2.4M org.seleniumhq.selenium.selenium-firefox-driver-2.39.0.jar
 57K org.seleniumhq.selenium.selenium-htmlunit-driver-2.39.0.jar
 12K org.seleniumhq.selenium.selenium-ie-driver-2.39.0.jar
 14K org.seleniumhq.selenium.selenium-iphone-driver-2.39.0.jar
377K org.seleniumhq.selenium.selenium-java-2.39.0.jar
241K org.seleniumhq.selenium.selenium-remote-driver-2.39.0.jar
272K org.seleniumhq.selenium.selenium-safari-driver-2.39.0.jar
132K org.seleniumhq.selenium.selenium-support-2.39.0.jar
 16K org.slf4j.jcl-over-slf4j-1.7.6.jar
4.8K org.slf4j.jul-to-slf4j-1.7.6.jar
 28K org.slf4j.slf4j-api-1.7.6.jar
109K org.specs2.classycle-1.4.3.jar
 67K org.specs2.specs2-analysis_2.11-2.3.12.jar
1.4M org.specs2.specs2-common_2.11-2.3.12.jar
2.0M org.specs2.specs2-core_2.11-2.3.12.jar
181K org.specs2.specs2-junit_2.11-2.3.12.jar
512K org.specs2.specs2-matcher-extra_2.11-2.3.12.jar
2.4M org.specs2.specs2-matcher_2.11-2.3.12.jar
223K org.specs2.specs2-mock_2.11-2.3.12.jar
 15K org.w3c.css.sac-1.3.jar
161K org.webbitserver.webbit-0.4.14.jar
272K xalan.serializer-2.7.1.jar
3.0M xalan.xalan-2.7.1.jar
1.3M xerces.xercesImpl-2.11.0.jar
215K xml-apis.xml-apis-1.4.01.jar
```

**Comment by akkie** [comment]
Yes, you're right. It should definitely be divided. I've created https://github.com/mohiva/play-silhouette-testkit.

Thanks

**Comment by rfranco** [comment]
I think it's possible to have both projects in the same git.
It's better to work

**Comment by akkie** [comment]
Indeed, but I have no idea how that could be achieved. Do you have an idea?

**Comment by rfranco** [comment]
OK, i'm going to do and submit a pull request, but i'll need to move files to new folder structure

**Comment by akkie** [comment]
OK, Please could you show the directory structure first?

**Comment by rfranco** [comment]
Here is the pull request #256

**Comment by akkie** [comment]
Closed with #256
